### ISSUE TITLE
npm module

### ISSUE THREAD

**Issue by Mitica** [silentmatt/expr-eval#29]
Will you publish a npm module?

**Comment by Spy-Seth** [comment]
:+1:

**Comment by jpravetz** [comment]
+1

**Comment by rreusser** [comment]
Here's an abandoned npm module: https://www.npmjs.com/package/node-expression-eval
And a somewhat heavily modified fork: https://www.npmjs.com/package/js-expression

It looks like there's still room for a nice clean mainstream module of this. @silentmatt if it works for you, I'd love to make the following changes and publish to npm:

- [x] fix 10/-1 parity issue (done?)
- [ ] add a linter (looks like one was used, but not explicitly enforced. no strong preference here—main goal is to catch messy things like unused/duplicate variables. I like [semistandard](https://github.com/Flet/semistandard) since it's zero-config and good enough, but I'm flexible)
- [ ] add package.json
- [ ] add travis.yml
- [ ] do the [AMD + CommonJS export thing](http://ifandelse.com/its-not-hard-making-your-library-support-amd-and-commonjs/)

Bottom line: math.js is like > 400K. [jsep](https://github.com/soney/jsep) isn't without issues and doesn't do super-simple evaluation. Esprima is overkill. So unless there's another candidate for sanitizing and evaluating expressions, seems like there's value in tightening things up and publishing it as a simple, robust expression evaluator.

**Comment by silentmatt** [comment]
@rreusser That sounds good to me. I pretty much have the same thoughts about semistandard, but I'm not sure about changing from tabs to spaces, only because it eliminates any git blame history. Although that might not be a huge deal, because for an old project there's not a huge amount of activity anyway :)

**Comment by rreusser** [comment]
Ah, `git blame -w`? Does that blame while ignoring whitespace? If so, that's pretty cool! http://stackoverflow.com/questions/3945382/git-commit-that-doesnt-override-original-authors-in-git-blame

**Comment by silentmatt** [comment]
That works for me then. I'll take a look at your pull request and probably merge it tonight after work.

**Comment by rreusser** [comment]
Cool. Really appreciated! Used it to make a pretty straightforward function plotter: http://rickyreusser.com/plotly-expression-transform/ (WIP. Unfortunately you need to double-click to get the change to have any effect. Still straightening that out.)

**Comment by silentmatt** [comment]
Nice :+1: That was actually my original use case for this.

**Comment by jonkemp** [comment]
Was this ever published to npm?

**Comment by silentmatt** [comment]
@jonkemp Not yet. I've been working on cleaning up the parser to make it clearer and easier to modify without breaking. I'm about done with that and I'm hoping to get that pushed out this weekend, and then publish it to npm.

**Comment by silentmatt** [comment]
It's actually been available in npm for a while now, but I'm closing this now that version 1.0.0 is out.
